optimade/server/tests/test_models.py

```diff
@@ -1,14 +1,14 @@
 import pytest
 from pydantic import BaseModel, ValidationError, ConfigError
 
-from .util import conlist
+from optimade.server.models.util import conlist
```

Comment:
Still need to figure out what we're doing with conlist... I tried recently with newer pydantic/FastAPI versions and still the constraint does not end up in the generated openapi spec.

Reply:
Ah - bad!
But I guess we can add `@validator`s for them, checking the length, etc.? That is quite straight forward, but a bit code-heavy and tedious.

Reply:
Actually, I think we should indeed at `@validator`s for the `StructureResourceAttributes` model, to make sure all the inter-field requirements are fulfilled etc.
I have started to do some of these, but far from all.

Reply:
I think there should be a proper discussion of this (again!) as the suggestion is was that the Python-generated openapi would be the ground truth schema (once it was complete), but I'm not sure these inter-field constraints are supported at all (of course, we could raise issues with fastapi or even contribute ourselves...). 

Do you have uncommitted code for these validators? Would be good to use them for my server implementation as a stop-gap solution.

Reply:
Can potentially hack the generated schema as described here: https://pydantic-docs.helpmanual.io/usage/schema/#schema-customization

Reply:
I have no idea about the truth-schema in openapi either. I would say the models themselves could act as truth schemas for Python implementations, while the generated openapi will simply be a JSON representation of it?

In the end, validation using pydantic's `@validator` is not difficult, just a bit tedious. I will try and do this.